src/hooks/useLocalization.ts

```diff
@@ -0,0 +1,12 @@
+import { buildPageTranslateFunction } from "../helpers";
+import { CustomTranslation } from "../types";
+import { useDictionary, useLanguage } from "./states"
+
+export const useLocalization = (customTranslations: CustomTranslation[] = []) => {
+  const [dictionary] = useDictionary();
+  const [language] = useLanguage();
+
```

Comment:
Why passing the custom translations as an argument why not adding it to the global state too

Reply:
I said that in the PR's description:
It's because the custom translations are a bunch of functions, and you cannot save function to state.

Reply:
Actually, I am not sure, if this is possible, maybe we can experiment with it.
If we could save a function with its closure to swr's global state, we can just save the `t` function, and make our hook to expose it.

Reply:
Ok then, this makes life a lot easier. Will push updates tonight

Reply:
@Doaa-Ismael @AhmedMedhatHabib Any updates?